paimon-spark/paimon-spark-common/src/main/scala/org/apache/spark/sql/Utils.scala

```diff
@@ -68,4 +75,41 @@ object Utils {
   def bytesToString(size: Long): String = {
     SparkUtils.bytesToString(size)
   }
+
+  // for analyze
+  def calculateTotalSize(
+      sessionState: SessionState,
+      tableName: String,
+      locationUri: Option[URI]): Long = {
+    CommandUtils.calculateSingleLocationSize(
+      sessionState,
+      new TableIdentifier(tableName),
+      locationUri)
+  }
+
+  def computeColumnStats(
+      sparkSession: SparkSession,
+      relation: LogicalPlan,
+      columns: Seq[Attribute]): (Long, Map[Attribute, ColumnStat]) = {
+    CommandUtils.computeColumnStats(sparkSession, relation, columns)
+  }
+
+  def analyzeSupportsType(dataType: DataType): Boolean = dataType match {
```

Comment:
Only these methods that are private in org.apache.spark.sql should be put here. Check these newly added ones.